src/os2datascanner/engine/utils.py

```diff
@@ -53,3 +59,37 @@ def get_data(file_name: str, uppercase=True) -> typing.FrozenSet[str]:
         for line in text.splitlines()
         if line
     )
+
+
+@contextmanager
+def prometheus_session(name, **kwargs):
+    while True:
+        # Find a free port to serve Prometheus metrics over...
+        try:
+            port = randrange(45000, 55000)
+            start_http_server(port)
+            break
+        except OSError as ex:
+            if ex.errno == errno.EADDRINUSE:
+                continue
+            else:
+                raise
+
+    # ... advertise this port, and this service, to Prometheus...
+    print(__file__)
+    advertisement_path = os.path.dirname(__file__) + "/../../../prometheus/{0}.json".format(name)
+    with open(advertisement_path, "w") as fp:
+        json.dump([
+            {
+                "targets": ["localhost:{0}".format(port)],
+                "labels": kwargs
+            }
+        ], fp)
+        fp.flush()
+        os.fsync(fp.fileno())
+
+    # ... yield...
```

Comment:
This comment doesn't really add anything, does it? 😁 
```suggestion
    # yield to execute whatever we're wrapping
```

Reply:
Fiddled around with in commit 0512599b6483d261e542aa47cb39161c5e2d055c.